### PR TITLE
data: add unit test for index retriever

### PR DESCRIPTION
**Pull request by amo33**
This PR adds test code checking whether retrieval process of a data sample is correct.
Output using ``make test`` :

```
Test project /home/flatflow/build
    Start 1: DatasetTest.Constructor
1/3 Test #1: DatasetTest.Constructor ...........   Passed    0.81 sec
    Start 2: DatasetTest.IntraBatchShuffling
2/3 Test #2: DatasetTest.IntraBatchShuffling ...   Passed    8.33 sec
    Start 3: DatasetTest.IndexRetriever
3/3 Test #3: DatasetTest.IndexRetriever ........   Passed    0.00 sec

100% tests passed, 0 tests failed out of 3

Total Test time (real) =   9.17 sec
```
Test log:

```
3/3 Testing: DatasetTest.IndexRetriever
3/3 Test: DatasetTest.IndexRetriever
Command: "/home/flatflow/build/flatflow/data/dataset_test" "--gtest_filter=DatasetTest.IndexRetriever"
Directory: /home/flatflow/build/flatflow/data
"DatasetTest.IndexRetriever" start time: Mar 28 02:10 UTC
Output:
----------------------------------------------------------
Running main() from /home/flatflow/third_party/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = DatasetTest.IndexRetriever
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from DatasetTest
[ RUN      ] DatasetTest.IndexRetriever
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1711591850.384404    8026 dataset.h:126] Construction of inverted index took 0.000854 seconds
[       OK ] DatasetTest.IndexRetriever (0 ms)
[----------] 1 test from DatasetTest (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 1 test.
<end of output>
Test time =   0.00 sec
----------------------------------------------------------
Test Passed.
"DatasetTest.IndexRetriever" end time: Mar 28 02:10 UTC
"DatasetTest.IndexRetriever" time elapsed: 00:00:00
----------------------------------------------------------

End testing: Mar 28 02:10 UTC

```

### CODE REVIEW

File: flatflow/data/dataset_test.cc [hunk 1]
```diff
@@ -179,4 +203,165 @@ TEST(DatasetTest, IntraBatchShuffling) {
   }
 }
 
+TEST(DatasetTest, IndexRetriever) {
+  // Note:
```
Comment (9rum) [inline flatflow/data/dataset_test.cc:207]: Please use upper case for comments to be rendered - i.e., ``NOTE:`` instead of ``Note:``.

**Review by 9rum — CHANGES_REQUESTED** [review]
@amo33 Thanks for filing this PR.
I have left a few comments, so please remind me after change.

**Comment by amo33** [comment]
@9rum I have updated the code to address all the changes you suggested:
- Use ternary operator 
- Fix comment style
- Rename function, arguments

Let me know if any further changes are needed.

File: flatflow/data/dataset_test.cc [hunk 2]
```diff
@@ -37,24 +37,48 @@ class DatasetTest final : private flatflow::data::Dataset<uint64_t, uint16_t> {
     return items.contains(size);
   }
 
-  inline std::size_t size(uint16_t size) const noexcept {
-    return items.at(size).size();
+  inline std::size_t size(uint16_t size, bool checkItem = true) const noexcept {
```
[the remark this answers is not in the document]
Comment (amo33) [reply flatflow/data/dataset_test.cc]: Since **_items_** is already used as class dataset's member variable, I changed checkItem to item.

**Review by 9rum — APPROVED** [review]
LGTM

**Pull request merged**